src/modules/esl-sortable/core/esl-sortable.ts

```diff
@@ -79,149 +80,154 @@ export class ESLSortable extends ESLBaseElement {
+
+    document.body.appendChild(this._$cloneEl.$host);
+    this.updateCloneElPosition(e);
   }
 
-  protected _onItemMove(e: any): void {
-    if (this.targetSortable?.transition || this.originSortable?.transition) return;
-
-    this._$targetElement = document.elementsFromPoint(e.x, e.y)[1] as any;
-    if (this.isSortableEmpty()) {
-      this._$targetElement.classList.add('esl-sortable-empty');
-      this.setTarget();
+  private findTargetItem(e: MouseEvent): void {
+    if (this.targetSortable?.transition || this.placeholderSortable?.transition) return;
+    const target = document.elementsFromPoint(e.x, e.y)[1] as ESLSortable;
+    this.placeholderSortable = this._$placeholderEl.parent!;
+    if (this.isSortableEmpty(target) && this.isSortableFromGroup(target) && this.placeholderSortable !== target) {
+      this._$targetEl = target;
+      this._$targetEl.empty = true;
+      this._$targetEl.$$cls('esl-sortable-target', true);
+      this.targetSortable = this._$targetEl;
+      this.fireInsertEvents(true);
       return;
     }
 
-    if (!this._$targetElement?.parentElement?.classList.contains('esl-sortable') ||
-        this._$targetElement.classList.contains('esl-sortable-drag') ||
-        this._$targetElement.classList.contains('esl-sortable-placeholder')) {
-      return;
-    }
-    this.setTarget();
+    const targetMixin = ESLSortableItem.get(target);
+    if (!targetMixin?.parent || targetMixin.placeholder) return;
+    this._$targetEl = targetMixin;
+    this.targetSortable = targetMixin.parent;
+    this.fireInsertEvents(false);
   }
 
-  private setTarget(): void {
-    this._$targetElement.classList.add('esl-sortable-target');
-    this.fireInsertEvents();
+  private fireInsertEvents(isSortableEmpty: boolean = false): void {
+    this._$targetEl.$$cls('esl-sortable-target', true);
+    if (!isSortableEmpty && !this.isSortableFromGroup(this.targetSortable)) return;
+    this.placeholderSortable.$$fire('sortable:insert', {detail: {target: this._$targetEl, placeholder: this._$placeholderEl, sameSortable: isSortableEmpty}});
+    // Fired when element was dragged from one sortable to another
+    if (this.targetSortable !== this.placeholderSortable) {
+      this.targetSortable.$$fire('sortable:insert', {detail: {target: this._$targetEl, placeholder: this._$placeholderEl, sameSortable: isSortableEmpty}});
+    }
   }
 
-  private fireInsertEvents(): void {
-    const targetParent = (this.isSortableEmpty() ? this._$targetElement :  this._$targetElement.parentNode) as ESLSortable;
-    const placeholderParent = this._$placeholderElement.parentNode as ESLSortable;
-    if (this._$targetElement.parentNode !== this && targetParent?.getAttribute('group') !== this.group) return;
-
-    placeholderParent.$$fire('sortable:insert', {detail: [this._$targetElement, this._$placeholderElement]});
+  /** Checks if target is sortable */
+  private isSortable(el: HTMLElement): boolean {
+    return el.classList.contains('esl-sortable');
```

Comment:
The name is hardcoded. Use something like that `const tagName = (this.constructor as typeof ESLBaseElement).is;`
This allows consumers to extend your component with its own functionality and name.